eth/fetcher/block_fetcher.go

```diff
@@ -729,12 +729,15 @@ func (f *BlockFetcher) ImportBlocks(peer string, block *types.Block, relay bool)
 	}
 
 	currentIntrinsicS := f.currentIntrinsicS()
-	maxAllowableEntropyDist := new(big.Int).Mul(currentIntrinsicS, big.NewInt(maxAllowableEntropyDist))
+	MaxAllowableEntropyDist := new(big.Int).Mul(currentIntrinsicS, big.NewInt(c_maxAllowableEntropyDist))
 
 	broadCastEntropy := block.ParentEntropy()
 
-	// If someone is mining not within maxAllowableEntropyDist*currentIntrinsicS
-	if relay && f.currentS().Cmp(new(big.Int).Add(broadCastEntropy, maxAllowableEntropyDist)) > 0 {
+	// If someone is mining not within MaxAllowableEntropyDist*currentIntrinsicS
+	if relay && f.currentS().Cmp(new(big.Int).Add(broadCastEntropy, MaxAllowableEntropyDist)) > 0 {
+		if nodeCtx != common.PRIME_CTX {
```

Comment:
Reasoning for this new prime exception?

Reply:
its expensive to drop peers in Prime, so we cannot at the moment, maybe in the future we can think of not having this exception